components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt

```diff
@@ -217,5 +221,6 @@ internal class SitePermissionsDialogFragment : AppCompatDialogFragment() {
             fragment.arguments = arguments
             return fragment
         }
+        lateinit var permissionRequest: PermissionRequest
```

Comment:
Won't this dialog last for a few seconds, until the user dismisses it or presses allow/don't allow?

Reply:
Yes, but if we hold the reference and Android kills the app and tries to restored it, this value will no be present and the app will crash when it tries to used it. To see this behaviour go developer options and activate "Don't keep activities", then trigger the dialog, with the dialog open and switch to another and come back to this app.

Reply:
While working on implementing your suggestion I decided to test how it works as is. Turns out with "Don't keep activities" enabled, the dialog fragment just gets destroyed and triggers the `onDismiss` thus consuming the permissionRequest which, I think, is a satisfactory behaviour. Did it work differently when you tested it?

Reply:
This behaviour is happening because `permissionRequest` is static it belongs to the companion object, holding static values is not a good idea as it could cause memory leaks and other issues. 
 
<img width="778" alt="image" src="https://user-images.githubusercontent.com/773158/96938699-159b8e80-1499-11eb-84d3-24bc969cff8e.png">